node/core/dispute-coordinator/src/real/ordering/mod.rs

```diff
@@ -132,19 +137,21 @@ impl OrderingProvider {
 		// Try to fetch the latest finalized block. Storing it in `last_observed_block`
 		// will enforce traversing non-finalized ancestors of `initial_head` in
 		// `process_active_leaves_update`.
-		let leaf = initial_head.hash;
-		let last_observed_block = get_finalized_block_hash(sender).await?.unwrap_or_else(|| {
+		let (block_number, hash) = get_finalized_block(sender).await?.unwrap_or_else(|| {
```

Comment:
This should really not be necessary, as in going back to the last finalized block should be the default if the LRU provided no hit, while traversing ancestry.